app/src/main/java/org/mozilla/fenix/library/bookmarks/BookmarkFragmentStore.kt

```diff
@@ -71,31 +74,37 @@ private fun bookmarkFragmentStateReducer(
             } + action.tree.guid
 
             val items = state.mode.selectedItems.filter { it in action.tree }
+            val mode = when {
+                state.mode is BookmarkFragmentState.Mode.Syncing -> {
+                    BookmarkFragmentState.Mode.Syncing
+                }
+                items.isEmpty() -> {
+                    BookmarkFragmentState.Mode.Normal(shouldShowMenu(action.tree.guid))
+                }
+                else -> BookmarkFragmentState.Mode.Selecting(items.toSet())
+            }
             state.copy(
                 tree = action.tree,
-                mode = when {
-                    state.mode is BookmarkFragmentState.Mode.Syncing -> {
-                        BookmarkFragmentState.Mode.Syncing
-                    }
-                    items.isEmpty() -> {
-                        BookmarkFragmentState.Mode.Normal(shouldShowMenu(action.tree.guid))
-                    }
-                    else -> BookmarkFragmentState.Mode.Selecting(items.toSet())
-                },
+                mode = mode,
                 guidBackstack = backstack,
-                isLoading = false
+                isLoading = false,
+                isSwipeToRefreshEnabled = mode !is BookmarkFragmentState.Mode.Selecting
             )
         }
-        is BookmarkFragmentAction.Select ->
-            state.copy(mode = BookmarkFragmentState.Mode.Selecting(state.mode.selectedItems + action.item))
+        is BookmarkFragmentAction.Select -> state.copy(
+                mode = BookmarkFragmentState.Mode.Selecting(state.mode.selectedItems + action.item),
+                isSwipeToRefreshEnabled = false
+            )
         is BookmarkFragmentAction.Deselect -> {
             val items = state.mode.selectedItems - action.item
+            val mode = if (items.isEmpty()) {
+                BookmarkFragmentState.Mode.Normal()
+            } else {
+                BookmarkFragmentState.Mode.Selecting(items)
+            }
             state.copy(
-                mode = if (items.isEmpty()) {
-                    BookmarkFragmentState.Mode.Normal()
-                } else {
-                    BookmarkFragmentState.Mode.Selecting(items)
-                }
+                mode = mode,
+                isSwipeToRefreshEnabled = mode is BookmarkFragmentState.Mode.Normal
```

Comment:
For `SwipeRefreshAvailabilityChanged` and `Change` I see we're checking `mode !is BookmarkFragmentState.Mode.Selecting`.
To be consistent maybe we should do the same check here?